cmd/main.go

```diff
@@ -23,59 +27,90 @@ func main() {
-		logger.Fatal("Failed to retrieve information about the Docker client and server host", err)
-	}
-	logger.Trace(fmt.Sprintf("Provider connection established with Docker %s (API %s)", serverVersion.Version, serverVersion.APIVersion))
+	if conf.ServiceDiscovery == config.Docker {
+		logger.Debug("Retrieving Docker server version")
+		serverVersion, err := docker.ServerVersion(conf.Docker)
+		if err != nil {
+			logger.Fatal("Failed to retrieve information about the Docker client and server host", err)
+		}
+		logger.Trace(fmt.Sprintf("Provider connection established with Docker %s (API %s)", serverVersion.Version, serverVersion.APIVersion))
 
+	}
 	logger.Info("Building Homer config from base config")
+
 	ctx := context.Background()
 	err = generateConfig(ctx, conf)
 	if err != nil {
 		logger.Fatal("Error generating Homer config", err)
 	}
 
-	logger.Info("Start watching for container creations and deletions")
-	eventsc, errc := docker.WatchEvents(ctx, conf.Docker)
-	for {
-		select {
-		case event := <-eventsc:
-			if event.Action == "start" || event.Action == "die" || strings.HasPrefix(event.Action, "health_status") {
-				logger.Trace(fmt.Sprintf("%+v", event))
-				logger.Debug("A " + event.Action + " event occurred")
-				logger.Info(fmt.Sprintf("Event '%s' received from %s. Generating Homer config...", event.Action, event.Actor.Attributes["name"]))
-				time.Sleep(1 * time.Second)
-				err = generateConfig(ctx, conf)
-				if err != nil {
-					logger.Fatal("Error generating Homer config", err)
+	if conf.ServiceDiscovery == config.Docker {
+		logger.Info("Start watching for container creations and deletions")
+		eventsc, errc := docker.WatchEvents(ctx, conf.Docker)
+		for {
+			select {
+			case event := <-eventsc:
+				if event.Action == "start" || event.Action == "die" || strings.HasPrefix(event.Action, "health_status") {
+					logger.Trace(fmt.Sprintf("%+v", event))
+					logger.Debug("A " + event.Action + " event occurred")
+					logger.Info(fmt.Sprintf("Event '%s' received from %s. Generating Homer config...", event.Action, event.Actor.Attributes["name"]))
+					time.Sleep(1 * time.Second)
+					err = generateConfig(ctx, conf)
+					if err != nil {
+						logger.Fatal("Error generating Homer config", err)
+					}
+				}
+			case err := <-errc:
+				if errors.Is(err, io.EOF) {
+					logger.Debug("Provider event stream closed")
 				}
+			case <-ctx.Done():
+				return
 			}
-		case err := <-errc:
-			if errors.Is(err, io.EOF) {
-				logger.Debug("Provider event stream closed")
+		}
+	} else if conf.ServiceDiscovery == config.Consul {
+		logger.Info("Start watching for consul services change")
+		hcLogger := hclog.New(&hclog.LoggerOptions{
+			Name:       "consulcatalog",
+			Level:      hclog.LevelFromString(conf.LogLevel),
+			JSONFormat: true,
+		})
+		for {
+			watcher := consul.WatchServices(conf.Consul)
+			watcher.HybridHandler = func(_ watch.BlockingParamVal, _ interface{}) {
+				logger.Info("consul handler fired")
```

Comment:
```suggestion
				logger.Info("Consul handler fired")
```

Can you change all logs to sentence case and use capitalise the C in Consul where appropriate?